src/app/signup/SignUpForm.tsx

```diff
@@ -0,0 +1,86 @@
+"use client";
+
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { signupUser } from "../../api/signupApi";
+import { useRouter } from "next/navigation";
+import Input from "@/components/inputs/Input";
+
+type SignupFormInputs = {
+  name: string;
+  email: string;
+  password: string;
+  passwordConfirm: string;
+};
+
+export function SignUpForm() {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors, isSubmitting },
+  } = useForm<SignupFormInputs>();
+  const [formError, setFormError] = useState("");
+  const router = useRouter();
+
+  const onSubmit = async (data: SignupFormInputs) => {
+    setFormError("");
+    try {
+      await signupUser(data);
+      router.push("/");
+    } catch (err: any) {
+      setFormError(err.message);
+    }
+  };
+
+  const password = watch("password");
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Input
+          id="name"
+          label="Name"
+          type="text"
+          {...register("name", { required: "Name is required" })}
+          error={errors.name?.message}
+        />
+        <Input
+          id="email"
+          label="Email"
+          type="email"
+          {...register("email", { required: "Email is required" })}
```

Comment:
It'd be a good idea to add a synchronized validation between the UI and the BE. You can e.g. create a `shared` directory in the project and put the validators there. This way, the UI will give a meaningful info about form issues and the back-end will actually validate the payload. It introduces redundancy, but assures that the UI returns meaningful messages and the BE is secure.